### PR TITLE
Enable ostree plugin also for upstream

### LINKED ISSUE #391: Upstream installer does not install plugins which are included in Satellite
According to our upstream automation and `standalone installer` jobs, installed katello (satellite-upstream) doesn't have plugins installed (discovery, libvirt, etc.) which causes our tests to fail.

I believe we consider `upstream` Katello with all plugins which are bundled together and installed by default in Satellite 6 therefore I'm marking this issue as a bug

### PR DESCRIPTION
**Pull request by lpramuk**
Enable ostree plugin also for upstream
- enable_ostree() refactored to handle plugin enablement also in upstream

(Fixes #391)

### CODE REVIEW

File: automation_tools/__init__.py [hunk 1]
```diff
@@ -1342,7 +1348,10 @@ def product_install(distribution, create_vm=False, certificate_url=None,
                 execute(setup_oscap, host=host)
             if satellite_version in ('6.2', ''):
                 execute(oscap_content, host=host)
-            execute(enable_ostree, host=host)
+        # ostree plugin is for Sat6.2+ rhel7 only
+        if satellite_version not in ('6.0', '6.1'):
```
Comment (omaciel) [inline automation_tools/__init__.py]: @lpramuk you'll have to rebase this PR, but as you do that, consider whether the `enable_ostree` task should be executed on `downstream`, `upstream` or both?

Comment (sghai) [inline automation_tools/__init__.py]: Should run on both but only on rhel7

File: automation_tools/__init__.py [hunk 2]
```diff
@@ -604,17 +604,23 @@ def setup_foreman_discovery():
...
     """
     os_version = distro_info()[1]
-    sat_version = os.environ.get('SATELLITE_VERSION')
-    # Enable ostree plugin only for Satellite6.2 installed on rhel7
-    if sat_version == '6.2' and os_version >= 7:
-        run('satellite-installer --scenario satellite '
-            '--katello-enable-ostree=true')
+    if os_version >= 7:
+        if upstream:
+            create_custom_repos(Centos_Atomic='http://buildlogs.centos.org/'
```
Comment (sghai) [inline automation_tools/__init__.py]: `s/Centos_Atomic/centos_atomic` ?

File: automation_tools/__init__.py [hunk 3]
```diff
@@ -604,17 +604,23 @@ def setup_foreman_discovery():
...
+                                'centos/7/atomic/x86_64/Packages/')
+        run('{}-installer --scenario {} --katello-enable-ostree=true'
+            .format(*['foreman', 'katello'] if upstream else ['satellite']*2))
+        if upstream:
+            delete_custom_repos('Centos_Atomic')
```
Comment (sghai) [inline automation_tools/__init__.py]: `s/Centos_Atomic/centos_atomic` ?

File: automation_tools/__init__.py (hunk 1 above)
Comment (lpramuk) [inline automation_tools/__init__.py]: not in ('6.0', '6.1') => 6.2 and nightly

File: automation_tools/__init__.py (hunk 2 above)
Comment (lpramuk) [inline automation_tools/__init__.py]: param name is translated to repo name.

**Comment by lpramuk** [comment]
Comments resolved.
Rebased.

Tempted to reverse logic for oscap_content task in favor of smarter 'not in old versions' to avoid adding new versions with every new release (6.3, 6.4...)

**Comment by sghai** [comment]
ACK

**Comment by rplevka** [comment]
ACK

**Pull request merged**